spec/features/merchant_index_page_spec.rb

```diff
@@ -27,5 +27,189 @@
+      user_1 = create(:user, city: 'Springfield', state: 'CO')
+      create(:fulfilled_order, user: user_1)
+      create(:fulfilled_order, user: user_1)
+      
+      user_2 = create(:user, city: 'Honolulu', state: 'HI')
+      4.times do
+        create(:fulfilled_order, user: user_2)
+      end
+      
+      user_3 = create(:user, city: 'Claremont', state: 'CA')
+      create(:fulfilled_order, user: user_3)
+      
+      user_4 = create(:user, city: 'New York City', state: 'NY')
+      
+      user_5 = create(:user, city: 'Honolulu', state: 'HI')
+      create(:fulfilled_order, user: user_5)
+      
+      user_6 = create(:user, city: 'Fairbanks', state: 'AK')
+      6.times do
+        create(:fulfilled_order, user: user_6, status: 0 )
+      end
+      
+      user_7 = create(:user, city: 'Fairbanks', state: 'AK')
+      6.times do
+        create(:fulfilled_order, user: user_7, status: 2)
+      end
+      
+      user_8 = create(:user, city: 'Springfield', state: 'MI')
+      create(:fulfilled_order, user: user_8)
+      create(:fulfilled_order, user: user_8)
+      create(:fulfilled_order, user: user_8)
+      
+      top_cities = "\n#{user_2.city}, #{user_2.state}\n#{user_8.city}, #{user_8.state}\n#{user_1.city}, #{user_1.state}"
+      
+      visit merchants_path 
+      
+      within "#statistics" do
+        expect(page).to have_content("Top 3 Cities with Most Orders:#{top_cities}")
+        expect(page).to_not have_content(user_4.city)
+        expect(page).to_not have_content(user_3.city)
+      end
+    end
+    it 'should show stats for the top 3 biggest order by quantity of items' do
+      order_1 = create(:fulfilled_order)
+      create(:fulfilled_order_item, order: order_1)
+      create(:fulfilled_order_item, order: order_1)
+      
+      order_2 = create(:fulfilled_order)
+      4.times do
+        create(:fulfilled_order_item, order: order_2)
+      end
+      
+      order_3 = create(:fulfilled_order)
+      3.times do
+        create(:fulfilled_order_item, order: order_3)
+      end
+      
+      order_4 = create(:fulfilled_order)
+      create(:fulfilled_order_item, order: order_4)
+      5.times do
+        create(:unfulfilled_order_item, order: order_4)
+      end
+      
+      order_5 = create(:order)
+      5.times do
+        create(:fulfilled_order_item, order: order_5)
+      end
+      
+      order_6 = create(:cancelled_order)
+      5.times do
+        create(:fulfilled_order_item, order: order_6)
+      end
+      
+      top_orders = "\nOrder ##{order_2.id}\nOrder ##{order_3.id}\nOrder ##{order_1.id}"
+      
+      visit merchants_path
+      
+      within "#statistics" do
+        expect(page).to have_content("Biggest Orders by Quantity of Items:#{top_orders}")
+        expect(page).to_not have_content("Order ##{order_4.id}")
+        expect(page).to_not have_content("Order ##{order_5.id}")
+        expect(page).to_not have_content("Order ##{order_6.id}")
+      end
+    end
```

Comment:
This looks really solid the only sad path I can think of is create an admin an unregistered user and see if they throw anything. Really nice work !

Reply:
I'll echo what Dan says here - tough to think of expectations for robustness as these are well-written. Splitting hairs would be testing that maybe non-enabled merchants don't show up? Solid testing!!